lib/Client.js

```diff
@@ -1242,8 +1276,8 @@ class Client extends EventEmitter {
                 };
             } else if(content.content !== undefined && typeof content.content !== "string") {
                 content.content = "" + content.content;
-            } else if(content.content === undefined && !content.embed && !content.embeds && content.flags === undefined) {
-                return Promise.reject(new Error("No content, embeds or flags"));
+            } else if(content.content === undefined && !content.embed && !content.embeds && !content.components && content.flags === undefined) {
```

Comment:
```suggestion
            } else if(content.content === undefined && !content.embed && (!content.embeds || content.embeds.length === 0) && (!content.components || content.components.length === 0) && content.flags === undefined) {
```

This should be added to avoid issues when people provide only an empty array.

Reply:
I think we can remove the check, there's an API error anyway and it's getting rather long

Reply:
The whole thing including for editMessage?

Reply:
Yeah, both create and edit. Nowadays this error is more likely to throw because of a new Discord feature than actually missing content.

Reply:
in my opinion, the error this provides looks and sounds a lot better than what Discord says, which is just extremely generic
> Cannot send an empty message

Reply:
Bring that up with discord then :-)